### PR TITLE
docs/ioc: expand Integration tests tutorial

### PR DESCRIPTION
**Pull request by minijackson**
Renamed to "Adding integration tests to your IOC", because an "Integration tests" user guide is planned, and to make it a bit clearer that it is a continuation of the StreamDevice tutorial.

Previously the tutorial was a mix between a guide and tutorial, not really guided enough.

Now the tutorial builds upon the StreamDevice tutorial, and shows a complete path to have a fully functional integration test.

Also introduce how to *run* tests, both non-interactively and interactively.

Added a quick reference of useful Python test functions.

The tutorial was also updated due to some changes in the default template, and some renamed options.

And some minor changes found during the rewriting of the tutorial.

### CODE REVIEW

**Comment by minijackson** [comment]
Note: beofre this gets merged, I need to figure out how to make this work for users of Nix inside a VM

In a lot of cases by default, KVM is not available, so they can't run the test VM (thanks @vivienCEA for reporting)

**Review by stephane-cea — APPROVED** [review]
Looks great to me! Thanks!

File: docs/ioc/tutorials/streamdevice.rst [hunk 1]
```diff
@@ -147,11 +147,11 @@ and what to expect in return.
    Terminator = LF;
 
    getVoltage {
-       out ":VOLT?"; in "%f";
+       out ":volt?"; in "%f";
```
Comment (agaget) [inline docs/ioc/tutorials/streamdevice.rst:150]: I'm surprised that it is now lower case. Usually SCPI commands are upper case.
Typically it's more :VOLTage or :VOLT? abbreviated
https://en.wikipedia.org/wiki/Standard_Commands_for_Programmable_Instruments

Comment (minijackson) [reply docs/ioc/tutorials/streamdevice.rst:150]: Yeah, that's something I might need to change in the simulator. In another PR?

Comment (minijackson) [reply docs/ioc/tutorials/streamdevice.rst:150]: On the SCPI-based power-supply that I controlled a while ago, it actually accepted both lower and uppercase. Documentation only mentioned uppercase, though. But I didn't find how to do it with Lewis.

Comment (agaget) [reply docs/ioc/tutorials/streamdevice.rst:150]: yeah sure

**Review by agaget — APPROVED** [review]

**Pull request merged**